### PR TITLE
Better SMS appt reminder checkbox error handling

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
This change adds properly handles a situation where a transaction cannot be created when a user clicks on the "appoints by SMS" checkbox. Previously a failure to create the transaction would fail silently and also give the impression that the change they requested had succeeded.

## Testing done
Enabled an existing test that had been disabled.
All other existing tests continue to pass.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/user/profile/vet360/components/base/Vet360Transaction.jsx [hunk 1]
```diff
@@ -22,15 +22,18 @@ function Vet360Transaction(props) {
   } = props;
 
   const method = transactionRequest?.method || 'PUT';
-  const hasError = isFailedTransaction(transaction);
+  const updateInProgress =
+    isPendingTransaction(transaction) || transactionRequest?.isPending;
+  const hasError =
+    isFailedTransaction(transaction) || transactionRequest?.isFailed;
```
Comment (sanlouise) [inline src/platform/user/profile/vet360/components/base/Vet360Transaction.jsx:28]: `isFailed` 😅

**Review by sanlouise — APPROVED** [review]
Looks good to me!

File: src/platform/user/profile/vet360/components/base/Vet360Transaction.jsx (hunk 1 above)
Comment (erikphansen) [reply src/platform/user/profile/vet360/components/base/Vet360Transaction.jsx:28]: oo I should double check this. we are using it elsewhere, but we also use `transactionRequest.error` 🤔

**Review by rianfowler — APPROVED** [review]

File: src/platform/user/profile/vet360/components/base/Vet360Transaction.jsx (hunk 1 above)
Comment (sanlouise) [reply src/platform/user/profile/vet360/components/base/Vet360Transaction.jsx:28]: Looking at [this test](src/platform/user/profile/vet360/tests/reducers/index.unit.spec.js), it seems we should have both `isFailed` and `error` available to us 😃 Not sure why there was ever a need to add `isFailed`.

**Pull request merged**